python_modules/dagster/dagster/_core/definitions/decorators/repository_decorator.py

```diff
@@ -118,7 +118,9 @@ def __call__(
         if isinstance(repository_definitions, list):
             bad_defns = []
             repository_defns = []
-            defer_repository_data = False
+            # If anv metadata is present, it is possible it contains cached info for defining
```

Comment:
We need to make this much more explicit at a minimum. But I don't understand why we need this at all? In our structure, if you never use `CacheableAssetsDefinition` you shouldn't have to create a pending repository at all.

Reply:
The short answer is that the simplest implementation for the goal of this PR made maximal use of the existing infrastructure, which runs through `PendingRepositoryDefinition`. The existing infrastructure uses `RepositoryLoadData` (which gets attached to `ExecutionPlanSnapshot`, which gets generated in the right place depending on whether there is a `PendingRepositoryDefinition`.

Getting away from `PendingRepositoryDefinition` will require additional complexity in this PR, which I would like to avoid for the MVP. `PendingRepositoryDefinition` is just an implementation detail here in any case.

Reply:
Yeah ok. Well as per your other comment we definitely need to have a more precise semantic than this in the form of an explicit metadata value type or something.

Reply:
Done, now works off detecting `CodeLocationStateMetadataValue`.

Reply:
What about something more explicit that 

`RequiresPendingRepositoryLoadingMetadataValue` or something dumb like that.